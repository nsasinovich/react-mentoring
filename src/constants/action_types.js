export default {
    CHANGE_SORT: 'CHANGE_SORT',
    UPDATE_SEARCH_INPUT: 'UPDATE_SEARCH_INPUT',
};
