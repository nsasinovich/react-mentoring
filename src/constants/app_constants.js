export const PosterSizes = {
    LARGE: 'poster-large',
    MEDIUM: 'poster-medium',
};

export const FilterOptions = {
    GENRE: 'genres',
    TITLE: 'title',
};

export const SortingOptions = {
    RELEASE: {
        name: 'release date',
        sortField: 'year',
    },
    RATING: {
        name: 'rating',
        sortField: 'vote_count',
    },
};

export const INITIAL_STATE = {
    results: [],
    resultsCount: 0,
    searchInput: '',
    selectedSort: SortingOptions.RATING,
    selectedFilter: FilterOptions.TITLE,
    selectedMovieDetails: null,
};
