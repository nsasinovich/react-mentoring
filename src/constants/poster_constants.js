export const PosterSizes = {
    LARGE: 'poster-large',
    MEDIUM: 'poster-medium',
};
