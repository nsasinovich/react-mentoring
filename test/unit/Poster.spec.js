import { shallow } from 'enzyme';
import React from 'react';
import { PosterSizes } from 'src/constants/app_constants';
import Poster from 'src/components/shared/poster/poster';

describe('<Poster />', () => {
    it('should render Poster and match snapshot', () => {
        const wrapper = shallow(<Poster />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Poster with image from url passed', () => {
        const posterUrl = 'https://cdn.miramax.com/media/assets/Pulp-Fiction1.png';
        const wrapper = shallow(<Poster imgUrl={posterUrl} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.props().children.props.src).toBe(posterUrl);
    });

    it('should render large Poster when large size is passed', () => {
        const wrapper = shallow(<Poster posterSize={PosterSizes.LARGE} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.props().className).toContain(PosterSizes.LARGE);
    });
});
