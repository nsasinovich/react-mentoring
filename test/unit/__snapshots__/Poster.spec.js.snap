// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Poster /> should render Poster and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Poster
    imgUrl="test-file-stub"
    posterSize="poster-medium"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        src="test-file-stub"
      />,
      "className": "poster poster-medium",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "src": "test-file-stub",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          src="test-file-stub"
        />,
        "className": "poster poster-medium",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "test-file-stub",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Poster /> should render Poster with image from url passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Poster
    imgUrl="https://cdn.miramax.com/media/assets/Pulp-Fiction1.png"
    posterSize="poster-medium"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        src="https://cdn.miramax.com/media/assets/Pulp-Fiction1.png"
      />,
      "className": "poster poster-medium",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "src": "https://cdn.miramax.com/media/assets/Pulp-Fiction1.png",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          src="https://cdn.miramax.com/media/assets/Pulp-Fiction1.png"
        />,
        "className": "poster poster-medium",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "https://cdn.miramax.com/media/assets/Pulp-Fiction1.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Poster /> should render large Poster when large size is passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Poster
    imgUrl="test-file-stub"
    posterSize="poster-large"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        src="test-file-stub"
      />,
      "className": "poster poster-large",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "src": "test-file-stub",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          src="test-file-stub"
        />,
        "className": "poster poster-large",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "test-file-stub",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
